utility/drivers.cpp

```diff
@@ -199,78 +159,190 @@ void l3gd20h_getRawTemperature(int8_t *pRawTemperature)
-    // 0b10000000 = 0x80 (+/- 4 gauss)
-    lsm.writeMagReg(LSM303::CRB_REG_M, 0x80);
-    _lsm303_d_mag_config.scale = LSM303_MAG_SCALE4GAUSS;
-
-    // Ctrl register for Data Output Rate & Temp sensor
-    // TEMP_EN 0 0 DO2 DO1 DO0 0 0
-    //    1    0 0  1   0   0  0 0
-    // 0x90 = 0b10010000 - temp enable, default data rate
-    lsm.writeMagReg(LSM303::CRA_REG_M, 0x90);
-  }
-}
-
-boolean lsm303_accel_init() {
+    // 0x64 = 0b01100100
+    // M_RES = 11 (high resolution mode); M_ODR = 001 (6.25 Hz ODR)
+    lsm.writeReg(LSM303::CTRL5, 0x64);
+
+    // Adjustable gauss scale:
+    //   MFS = 0b00000000 (+/-  2 gauss full scale) = 0x00
+    //   MFS = 0b00100000 (+/-  4 gauss full scale) = 0x20 (Default)
+    //   MFS = 0b01000000 (+/-  8 gauss full scale) = 0x40
+    //   MFS = 0b01100000 (+/- 12 gauss full scale) = 0x60
+    switch (gaussScale) {
+      case LSM303_MAG_SCALE1_3GAUSS:
+        _lsm303_d_mag_config.scale = LSM303_MAG_SCALE2GAUSS;
+      case LSM303_MAG_SCALE2GAUSS:
+        scale = 0x00;
+        break;
+      case LSM303_MAG_SCALE2_5GAUSS:
+        _lsm303_d_mag_config.scale = LSM303_MAG_SCALE4GAUSS;
+      case LSM303_MAG_SCALE4GAUSS:
+        scale = 0x20;
+        break;
+      case LSM303_MAG_SCALE4_7GAUSS:
+        _lsm303_d_mag_config.scale = LSM303_MAG_SCALE8GAUSS;
+      case LSM303_MAG_SCALE5_6GAUSS:
+        _lsm303_d_mag_config.scale = LSM303_MAG_SCALE8GAUSS;
+      case LSM303_MAG_SCALE8GAUSS:
+        scale = 0x40;
+        break;
+      case LSM303_MAG_SCALE12GAUSS:
+        scale = 0x60;
+        break;
+    }
+    lsm.writeReg(LSM303::CTRL6, scale);
+
+    // 0x00 = 0b00000000
+    // MLP = 0 (low power mode off); MD = 00 (continuous-conversion mode)
+    lsm.writeReg(LSM303::CTRL7, 0x00);
+  }
+  else // LSM303 DLHC (Breakout Space Kit variant)
+  {
+    // 0x0C = 0b00001100
+    // DO = 011 (7.5 Hz ODR)
+    lsm.writeMagReg(LSM303::CRA_REG_M, 0x0C);
+
+    // Adjustable gauss scale:
+    //   MFS = 0b00100000 (+/- 1.3 gauss full scale) = 0x20
+    //   MFS = 0b01000000 (+/- 1.9 gauss full scale) = 0x40
+    //   MFS = 0b01100000 (+/- 2.5 gauss full scale) = 0x60
+    //   MFS = 0b10000000 (+/- 4.0 gauss full scale) = 0x80 (Default)
+    //   MFS = 0b10100000 (+/- 4.7 gauss full scale) = 0xA0
+    //   MFS = 0b11000000 (+/- 5.6 gauss full scale) = 0xC0
+    //   MFS = 0b11100000 (+/- 8.1 gauss full scale) = 0xE0
+    switch (gaussScale) {
+      case LSM303_MAG_SCALE1_3GAUSS:
+        scale = 0x20;
+        break;
+      case LSM303_MAG_SCALE2GAUSS:
+        scale = 0x40;
+        break;
+      case LSM303_MAG_SCALE2_5GAUSS:
+        scale = 0x60;
+        break;
+      case LSM303_MAG_SCALE4GAUSS:
+        scale = 0x80;
+        break;
+      case LSM303_MAG_SCALE4_7GAUSS:
+        scale = 0xA0;
+        break;
+      case LSM303_MAG_SCALE5_6GAUSS:
+        scale = 0xC0;
+        break;
+      case LSM303_MAG_SCALE12GAUSS:
+        _lsm303_d_mag_config.scale = LSM303_MAG_SCALE8GAUSS;
```

Comment:
It seems like we should always assign this variable, ya? No matter what, add some comments explaining all this custom roll-through logic and sometimes assigning it, sometimes not...